addons/l10n_latam_check/models/account_payment.py

```diff
@@ -1,328 +1,252 @@
+    @api.depends('l10n_latam_move_check_ids.amount', 'l10n_latam_new_check_ids.amount', 'payment_method_code')
+    def _compute_amount(self):
         for rec in self:
-            rec.l10n_latam_check_number = rec.check_number
+            checks = rec.l10n_latam_new_check_ids if rec._is_latam_check_payment(check_subtype='new_check') else rec.l10n_latam_move_check_ids
+            if checks:
+                rec.amount = sum(checks.mapped('amount'))
+
+    def _is_latam_check_payment(self, check_subtype=False):
+        if check_subtype == 'move_check':
+            codes = ['in_third_party_checks', 'out_third_party_checks']
+        elif check_subtype == 'new_check':
+            codes = ['new_third_party_checks', 'own_checks']
+        else:
+            codes = ['in_third_party_checks', 'out_third_party_checks', 'new_third_party_checks', 'own_checks']
+        return self.payment_method_code in codes
 
-    def _inverse_l10n_latam_check_number(self):
-        for rec in self:
-            rec.check_number = rec.l10n_latam_check_number
-
-    def _compute_check_number(self):
-        """ Override from account_check_printing.
-        For electronic/deferred own checks or third party checks, don't call super so that number is not cleaned """
-        latam_checks = self.filtered(
-            lambda x: x.payment_method_line_id.code == 'new_third_party_checks' or
-            (x.payment_method_line_id.code == 'check_printing' and x.l10n_latam_manual_checks))
-        return super(AccountPayment, self - latam_checks)._compute_check_number()
-
-    def _inverse_check_number(self):
-        """ On third party checks or electronic/deferred own checks, avoid calling super because is not needed to write
-        the sequence for these use case. """
-        avoid_inverse = self.filtered(
-            lambda x: x.l10n_latam_manual_checks or x.payment_method_line_id.code == 'new_third_party_checks')
-        return super(AccountPayment, self - avoid_inverse)._inverse_check_number()
-
-    @api.depends('payment_method_line_id.code', 'partner_id')
-    def _compute_l10n_latam_check_bank_id(self):
-        new_third_party_checks = self.filtered(lambda x: x.payment_method_line_id.code == 'new_third_party_checks')
-        for rec in new_third_party_checks:
-            rec.l10n_latam_check_bank_id = rec.partner_id.bank_ids[:1].bank_id
-        (self - new_third_party_checks).l10n_latam_check_bank_id = False
+    def action_post(self):
+        # unlink checks if payment method code is not for checks. We do it on post and not when changing payment
+        # method so that the user don't loose checks data in case of changing payment method and coming back again
+        # also, changing partner recompute payment method so all checks would be cleaned
+        for payment in self.filtered(lambda x: x.l10n_latam_new_check_ids and not x._is_latam_check_payment(check_subtype='new_check')):
```

Comment:
I had put comments here and I do not understand how you managed to have them gone...  (the same ones to unify the 2 check methods)

Reply:
mmm, maybe an they where on an old pr?

Reply:
Done